### PR TITLE
[ARTS-408] Run API Tests in GitHub CI

### PR DESCRIPTION
**Pull request by mdp0**
## Description
Run Mocha tests against docker compose in build env

### Changes
- [ARTS-408] - Run api tests, report results
- Add new services into check-docker-compose-services
- Fix health checks for new services

### Checklist:

- [x] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-408]: https://ndph-arts.atlassian.net/browse/ARTS-408

### CODE REVIEW

**Review by tamirkamara — APPROVED** [review]

**Review by shajidissa — APPROVED** [review]
lovely bit of docker. 👍

File: .ci/docker/check-docker-compose-services.sh [hunk 1]
```diff
@@ -15,5 +17,7 @@ is_healthy() {
     fi
 }
 
-while ! is_healthy trial-config-service; do sleep 5; done
-while ! is_healthy practitioner-service; do sleep 5; done
\ No newline at end of file
+while ! is_healthy config-server; do sleep 10; done
+while ! is_healthy site-service; do sleep 10; done
+while ! is_healthy practitioner-service; do sleep 10; done
+while ! is_healthy role-service; do sleep 10; done
```
Comment (darrengsmithson) [inline .ci/docker/check-docker-compose-services.sh:23]: add echo "Services running..."  at the eof?

File: api-tests/specs/role-service/roleservice.js [hunk 2]
```diff
@@ -1,39 +1,36 @@
 const requests = require('../../data/role-service/roleservice')
 const conf = require('../../config/conf')
-
-beforeEach(function () {
-    baseRequest = request(conf.BASE_URL)
-})
+const endpointUri = ':82/roles';
```
Comment (darrengsmithson) [inline api-tests/specs/role-service/roleservice.js:3]: is there a way of not hard-wiring the port number?

Comment (mdp0) [reply api-tests/specs/role-service/roleservice.js:3]: Left that as an exercise for QA-  have chatted to Sameera about it. Can remove these when we get the gateway in.

**Review by tamirkamara — DISMISSED** [review]

File: .ci/docker/check-docker-compose-services.sh (hunk 1 above)
Comment (mdp0) [reply .ci/docker/check-docker-compose-services.sh:23]: Added a success message

**Review by tamirkamara — APPROVED** [review]

**Review by darrengsmithson — APPROVED** [review]

**Pull request merged**